### PR TITLE
Adding Windows builds and compiles.

### PR DESCRIPTION
**Pull request by phillipsj**
We need to take this step to start working on getting the system agent running on Windows.

Signed-off-by: Jamie Phillips <jamie.phillips@suse.com>

### CODE REVIEW

**Review by Oats87 — APPROVED** [review]
LGTM

**Comment by Oats87** [comment]
@briandowns can you also take a look at this?

File: pkg/config/config_unix.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+package config
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+func pathOwnedByRoot(fi os.FileInfo, path string) error {
+	if fi.Sys().(*syscall.Stat_t).Uid != 0 || fi.Sys().(*syscall.Stat_t).Gid != 0 {
```
Comment (briandowns) [inline pkg/config/config_unix.go:12]: Can you perform the type cast in the form of `if v, ok := whatever` and check the `ok` value?

File: scripts/version [hunk 2]
```diff
@@ -6,7 +6,7 @@ fi
 
 COMMIT=$(git rev-parse --short HEAD)
 GIT_TAG=${DRONE_TAG:-$(git tag -l --contains HEAD | head -n 1)}
-
+CROSS=true
```
Comment (briandowns) [inline scripts/version:9]: Is this value used elsewhere?

File: pkg/config/config_unix.go (hunk 1 above)
Comment (phillipsj) [reply pkg/config/config_unix.go:12]: This was the existing code and I just moved it. Happy to make the changes.

File: scripts/version (hunk 2 above)
Comment (phillipsj) [reply scripts/version:9]: That was already configured in the build script to enable cross compilation. I just used the existing hook to make it trigger.

File: pkg/config/config_unix.go (hunk 1 above)
Comment (briandowns) [reply pkg/config/config_unix.go:12]: Sorry. If you dont' mind. It'd be safer with the change.

**Review by briandowns — APPROVED** [review]

**Pull request merged**